### PR TITLE
[GSOC] integrated support for Zvbc

### PR DESCRIPTION
**Pull request by SyedHassanUlHaq**

### CODE REVIEW

File: generator/insn_vdvs2rs1vm.go [hunk 1]
```diff
@@ -9,12 +9,15 @@ import (
 func (i *Insn) genCodeVdVs2Rs1Vm(pos int) []string {
 	vdWidening := strings.HasPrefix(i.Name, "vw")
 	vs2Widening := strings.HasSuffix(i.Name, ".wx")
+	sew64_insn := i.Name == "vclmul.vx" || i.Name == "vclmulh.vx"
```
Comment (ksco) [inline generator/insn_vdvs2rs1vm.go:12]:
```suggestion
	sew64Only := strings.HasSuffix(i.Name, "vclmul")
```

File: generator/insn_vdvs2rs1vm.go [hunk 2]
```diff
@@ -9,12 +9,15 @@ import (
...
 	vdSize := iff(vdWidening, 2, 1)
 	vs2Size := iff(vs2Widening, 2, 1)
 
 	sews := iff(vdWidening || vs2Widening, allSEWs[:len(allSEWs)-1], allSEWs)
+	sews = iff(sew64_insn, []SEW{64}, sews)
+
 	combinations := i.combinations(
-		iff(vdWidening || vs2Widening, wideningMULs, allLMULs),
+		iff(vdWidening || vs2Widening, wideningMULs, iff(sew64_insn, []LMUL{1, 2, 4, 8}, allLMULs)),
```
Comment (ksco) [inline generator/insn_vdvs2rs1vm.go:20]: So, IIUC, for like VLEN=128, LMUL=1/2 should be okay?

**Review by ksco — CHANGES_REQUESTED** [review]

File: generator/insn_vdvs2rs1vm.go (hunk 2 above)
Comment (SyedHassanUlHaq) [reply generator/insn_vdvs2rs1vm.go:20]: spike returned illegal instruction trap for all fractional LMUL

Comment (ksco) [reply generator/insn_vdvs2rs1vm.go:20]: Okay, but can you explain why?

Comment (SyedHassanUlHaq) [reply generator/insn_vdvs2rs1vm.go]: Chapter 4, Section 2 of the riscv vector spec specifies that LMUL can not be less then 1 when SEW=64

Comment (ksco) [reply generator/insn_vdvs2rs1vm.go]: image
Are you sure? I see no similar statements in 4.2.

**Comment by ksco** [comment]
Sorry, why this is closed?

**Comment by SyedHassanUlHaq** [comment]
> Sorry, why this is closed?

My GSOC mentor Zewen told me that we are not going to support some instructions with SEW=64.
So we can skip these SEW=64 instructions and I can close the PR

**Comment by ksco** [comment]
Okay, but the PR is good, so why not merge it? I just need an explanation about valid SEW values.

**Comment by SyedHassanUlHaq** [comment]
image

here it states that LMUL >= SEWmin/ELEN. So for a instruction that only supports SEW=64, SEWmin=64 and ELEN=64
Hence LMUL >= 1

**Comment by ksco** [comment]
I see, can you make another PR so we can merge this in?

**Comment by SyedHassanUlHaq** [comment]
sure

**Pull request closed without merging**